src/main/java/uk/gov/pay/adminusers/resources/UserResource.java

```diff
@@ -79,6 +80,30 @@ public Response getUser(@PathParam("externalId") String externalId) {
                 .orElseGet(() -> Response.status(NOT_FOUND).build());
     }
 
+    @Path(USER_EMAILS_RESOURCE)
+    @POST
+    @Produces(APPLICATION_JSON)
+    @Consumes(APPLICATION_JSON)
+    public Response getUserEmails(JsonNode payload) {
+        if (payload != null && payload.get(GetUserProfilesRequest.FIELD_EXTERNAL_IDS) != null) {
+            logger.info("Get UserProfiles request - [ {} ]", payload.toString());
+
+            List<String> externalIds = newArrayList(payload.get(GetUserProfilesRequest.FIELD_EXTERNAL_IDS).elements())
+                    .stream().map(externalIdNode -> externalIdNode.textValue()).collect(Collectors.toList());
+
+            List<UserEmail> userEmails = userServices.findUserByExternalIds(externalIds);
+
+            return Response.status(OK).type(APPLICATION_JSON).entity(
+                            Collections.unmodifiableMap(Stream
+                                       .of(new AbstractMap.SimpleEntry<String, List<UserEmail>>("results", userEmails))
+                                       .collect(Collectors.toMap((e) -> e.getKey(), (e) -> e.getValue()))
```

Comment:
The brackets around each `e` aren’t really gaining us much and could probably be removed